tests/unittest_stralloc.c

```diff
@@ -0,0 +1,48 @@
+#include <check.h>
+
+#include "stralloc.h"
+
+START_TEST(test_stralloc_thingy)
+{
+  stralloc thingy = {0};
+  const char *input = "thingy";
+
+  stralloc_copys(&thingy,input);
+  stralloc_0(&thingy);
+
+  ck_assert_str_eq(input, thingy.s);
+}
```

Comment:
This leaks the contents of thingy.

Reply:
You're right. Initially I wasn't sure this was worth doing, because in the entire source tree nearly every `sa.s` relies on the process to end, rather than explicitly calling `alloc_free()` when the stralloc goes out of scope. But as @DerDakon reminded me in IRC, we know we'll want to run tests under `valgrind`.

While I'm mentioning @DerDakon, I've neglected to mention that in this revised PR submission the `tests` subdir with its own `Makefile` was provided by @DerDakon. Thanks :-)